mrmustard/lab_dev/wires.py

```diff
@@ -108,110 +119,158 @@ def __init__(
+        The set of modes of the populated wires.
+        """
         return [m for m in self._modes if any(self.id_array[self._modes.index(m)] > 0)]
 
     @property
     def indices(self) -> list[int]:
-        r"""Returns the array of indices of this subset in the standard order.
-        (bra/ket x out/in x mode). Use this to get the indices for bargmann contractions.
+        r"""
+        The array of indices of this subset in the standard order.
         """
         flat = self.id_array.T.ravel()
         flat = flat[flat != 0]
         return np.where(flat > 0)[0].tolist()
 
     @property
     def input(self) -> Wires:
-        "A view of self without output wires"
+        r"""
+        A view of this ``Wires`` object without output wires.
+        """
         return self._view(masked_cols=(0, 2))
 
     @property
     def output(self) -> Wires:
-        "A view of self without input wires"
+        r"""
+        A view of this ``Wires`` object without input wires.
+        """
         return self._view(masked_cols=(1, 3))
 
     @property
     def ket(self) -> Wires:
-        "A view of self without bra wires"
+        r"""
+        A view of this ``Wires`` object without bra wires.
+        """
         return self._view(masked_cols=(0, 1))
 
     @property
     def bra(self) -> Wires:
-        "A view of self without ket wires"
+        r"""
+        A view of this ``Wires`` object without ket wires.
+        """
         return self._view(masked_cols=(2, 3))
 
     @property
     def adjoint(self) -> Wires:
         r"""
-        The adjoint (ket <-> bra) of this wires object.
+        The adjoint of this wires object, obtained by swapping ket and bra wires, with new ids.
         """
         return self._from_data(self._id_array[:, [2, 3, 0, 1]], self._modes, self._mask)
 
     @property
     def dual(self) -> Wires:
         r"""
-        The dual (in <-> out) of this wires object.
+        The dual of this wires object, obtained by swapping input and output wires, with new ids.
         """
         return self._from_data(self._id_array[:, [1, 0, 3, 2]], self._modes, self._mask)
 
     def copy(self) -> Wires:
-        r"""A copy of this Wires object with new ids."""
+        r"""
+        A copy of this wire with new ids.
+        """
         w = Wires(*self._args())
         w._mask = self._mask.copy()
         return w
 
+    def __add__(self, other: Wires) -> Wires:
+        r"""
+        A new ``Wires`` object with the wires of self and other combined.
+        """
+        modes_rows = {}
+        all_modes = sorted(set(self.modes) | set(other.modes))
+        for m in all_modes:
+            self_row = self.id_array[self._modes.index(m)] if m in self.modes else np.zeros(4)
+            other_row = other.id_array[other._modes.index(m)] if m in other.modes else np.zeros(4)
+            if np.any(np.where(self_row > 0) == np.where(other_row > 0)):
+                raise ValueError(f"wires overlap on mode {m}")
```

Comment:
Are we going to mention the raiseErrors in the docs or not?

Reply:
Thanks for spotting this!